src/main/java/org/helpberkeley/memberdata/HBParser.java

```diff
@@ -606,17 +559,32 @@ static void parseDetails(long postNumber, final String rawPost,
     }
 
     public static String shortURLDiscoursePost(final String line) {
-        int index = line.indexOf(Constants.UPLOAD_URI_PREFIX);
+        UploadFile.auditFilePrefix(line);
+        int index = -1;
+        int prefixLength = 0;
+        if (line.contains(Constants.UPLOAD_URI_PREFIX)) {
+            index = line.indexOf(Constants.UPLOAD_URI_PREFIX);
+            prefixLength= Constants.UPLOAD_URI_PREFIX.length();
+        }
+        else if (line.contains(Constants.WEB_CSV_PREFIX)){
+            index = line.indexOf(Constants.WEB_CSV_PREFIX);
+            prefixLength= Constants.WEB_CSV_PREFIX.length();
+        }
         assert index != -1 : line;
-        String shortURL = line.substring(index);
+        String shortURL = Constants.UPLOAD_URI_PREFIX.concat(line.substring(index + prefixLength));
         index = shortURL.indexOf(')');
         shortURL = shortURL.substring(0, index);
 
         return shortURL;
     }
 
     static String shortURLUploadResponse(final String line) {
-        int index = line.indexOf(Constants.UPLOAD_URI_PREFIX);
+        UploadFile.auditFilePrefix(line);
+        int index = -1;
+        if(line.contains(Constants.UPLOAD_URI_PREFIX))
+            index = line.indexOf(Constants.UPLOAD_URI_PREFIX);
+        else if (line.contains(Constants.WEB_CSV_PREFIX))
+            index = line.indexOf(Constants.WEB_CSV_PREFIX);
         assert index != -1 : line;
         String shortURL = line.substring(index);
         index = shortURL.indexOf('"');
```

Comment:
This is doing some hand-parsing of a JSON response to an upload done by the software.  This would cleaner if we did our regular JSON parsing on it and created an UploadResponse object.   Then we would use that to construct a new UploadFile object.  How about I add parsing for UploadResponse and then you can add UploadFile.createUploadFile(UploadResponse)?   I think that this would better hide the parsing, and normalize higher level usage of this by always dealing with UploadFile instances in the higher level code.